File: sample-example-bellman.go
package main

import (
	"crypto/rand"
	"fmt"

	"github.com/nadavgolan/zk-snarks"
)

func main() {
	// Define the statement we want to prove
	statement := []byte("This is the statement we want to prove")

	// Generate a random proof
	proof, err := zk_snarks.Prove(statement, rand.Reader)
	if err != nil {
		fmt.Println(err)
		return
	}

	// Verify the proof
	err = zk_snarks.Verify(statement, proof)
	if err != nil {
		fmt.Println(err)
		return
	}

	fmt.Println("Proof verified successfully")
}

File: sample-example-bellman.rs
use bellman::groth16::{create_random_proof, prepare_verifying_key, verify_proof};
use bellman::{Circuit, ConstraintSystem, SynthesisError};
use rand::rngs::OsRng;

// Define the circuit that represents the statement we want to prove
struct MyCircuit;

impl Circuit<bellman::bellman::Scalar> for MyCircuit {
  fn synthesize<CS: ConstraintSystem<bellman::bellman::Scalar>>(
    self,
    cs: &mut CS,
  ) -> Result<(), SynthesisError> {
    // Implement the circuit here...

    Ok(())
  }
}

fn main() {
  // Generate a random proof
  let mut rng = OsRng::new().unwrap();
  let (proof, vk) = create_random_proof(MyCircuit, &mut rng).unwrap();

  // Verify the proof
  let pvk = prepare_verifying_key(&vk);
  assert!(verify_proof(&pvk, &proof, &[]).unwrap());
}
